### ISSUE TITLE
Feature suggestion: `strict` parameter for `zip`

### ISSUE THREAD

**Issue by smheidrich** [vxgmichel/aiostream#118]
Python 3.10 introduced the `strict` parameter for the [built-in `zip` function](https://docs.python.org/3/library/functions.html#zip), which, when set to `True`, can help spot bugs by raising an exception when the supplied iterables turn out to have different lengths (i.e. when one raises `StopIteration` after fewer iterations than the others). 

I think it would be useful if [aiostream's `zip` function](https://aiostream.readthedocs.io/en/latest/operators.html#aiostream.stream.zip) supported this parameter as well.

Extra context: The `strict` parameter of Python's built-in `zip` turned out to be useful enough that [linters like Ruff or flake8 with the bugbear plugin even have a rule to complain](https://docs.astral.sh/ruff/rules/zip-without-explicit-strict/) when you neither use it nor explicitly disable it by setting `strict=False`.

**Comment by vxgmichel** [comment]
Hi @smheidrich, thanks for the suggestion :+1: 

I think it would be useful too, feel free to create a PR if you feel like contributing to the project. The implementation is over there:
https://github.com/vxgmichel/aiostream/blob/5e0547afcebfcb556e011df3c185f69794a187cb/aiostream/stream/combine.py#L49-L86

It relies on [asyncio.gather](https://docs.python.org/3/library/asyncio-task.html#asyncio.gather) which exposes a `return_exceptions` argument. However, I don't think we can use that since we still want to propagate the exception as soon as possible if it is not a `StopAsyncIteration`. Instead, we probably want to turn `StopAsyncIteration` into a sentinel with a dedicated function:

```python
async_stop_iteration_sentinel = object()
async def next_item(streamer):
    try:
        return await anext(streamer)
    except StopAsyncIteration:
        return async_stop_iteration_sentinel 


[...]
coros = builtins.map(next_item, streamers)
items = await asyncio.gather(*coros)
if any(item == async_stop_iteration_sentinel for item in items):
    if strict and not all(item == async_stop_iteration_sentinel for item in items)
        raise ValueError
    break
```

**Comment by vxgmichel** [comment]
Done and released in [v0.6.2](https://github.com/vxgmichel/aiostream/releases/tag/v0.6.2) :tada: 

Thanks a lot @smheidrich for you valuable contribution !

**Comment by smheidrich** [comment]
Great!! Thanks for the reviews & merge! :+1:
